src/app/(shared-nav)/courses/page.tsx

```diff
@@ -0,0 +1,101 @@
+"use client"
+import { languages } from "@/app/enums/language.enum";
+import { levels } from "@/app/enums/level.enum";
+import { CoursCard } from "@/components/coursCard";
+import { coursDetails } from "@/types/coursDetails.type";
+import { ChangeEvent, useEffect, useState } from "react";
+import { IoSearch } from "react-icons/io5";
+
+export default function Courses(){
+    const [searchText, setSearchText] = useState<String>("");
+    const [courses, setCourses] = useState<coursDetails[]>(DATA);
+    const handleSearch = (event: ChangeEvent<HTMLInputElement>)=>{
+        setSearchText(event.target.value);
+    }
+    useEffect(()=>{
+        if(searchText !== "")
+            setCourses(DATA.filter(cours=> cours.title.includes(searchText as string)))
+        else if([null, undefined, ""].includes(searchText as (string | null | undefined))){
+            //force to set the value of searchText state to be an empty string even the if the stored value is undefined or null
+            setSearchText("");
+            setCourses(DATA);
+        }
+    }, [searchText]);
+    const handleSuggestion = (title: string)=>{
+        setSearchText(title);
+    }
+    return <div className="text-black w-screen flex-1 flex justify-start items-center flex-col gap-10 pt-10">
+        {/* search input */}
+        <div className="p-2 border-[2px] border-black rounded-lg flex justify-center items-center gap-3 w-[340px]">
+            <input placeholder="برمجة"
+                   type="text"
+                   className="text-center text-black font-bold placeholder-black focus:outline-none"
+                   onChange={handleSearch}
+                   value={searchText as string}
+            />
+            <IoSearch className="w-[25px] h-[25px] font-bold"/>
+        </div>
+        {/* best keywords */}
+        <div className="flex flex-col gap-5 items-end">
+            <p>متداول الان</p>
+            <div className="flex gap-3">
+                <button onClick={()=>handleSuggestion("تعلم الآلة")} className="border border-black py-2 px-4 font-bold bg-[#FF0000] text-white rounded-md">تعلم الآلة</button>
+                <button onClick={()=>handleSuggestion("الخوارزميات")} className="border border-black py-2 px-4 font-bold bg-[#FF0000] text-white rounded-md">الخوارزميات</button>
+                <button onClick={()=>handleSuggestion("اساسيات البرمجة")} className="border border-black py-2 px-4 font-bold bg-[#FF0000] text-white rounded-md">اساسيات البرمجة</button>
+            </div>
+        </div>
+        <div></div>
+        {/* list of courses */}
+        <div className="flex justify-center items-center gap-5 flex-wrap">
+            {courses.map((data)=><CoursCard data={data}/>)} 
```

Comment:
You should add a key here. You could even use the index like this {courses.map((data, index)=><CoursCard key={index} data={data}/>)}

Reply:
sorry for that, i forgot to add the key in this iteration,
is it okay to work with uuid, since all the the mapping that I've worked one used uuid knowing that we don't have an id yet for the data and also working with the index provided by `.map` can cause some duplication problems